### ISSUE TITLE
JS runtime error while inspecting memtrace 

### ISSUE THREAD

**Issue by toots** [janestreet/memtrace_viewer#12]
Happy to provide the corresponding trace.

<img width="709" alt="Screenshot 2024-03-27 at 10 25 12 AM" src="https://github.com/janestreet/memtrace_viewer/assets/871060/e559d0c8-7caf-4889-b24c-d2b2a32e7ab3">

<img width="291" alt="Screenshot 2024-03-27 at 10 25 22 AM" src="https://github.com/janestreet/memtrace_viewer/assets/871060/bd13c930-250d-475c-a154-21bbeb22a495">

**Comment by toots** [comment]
This appears to be a regression -- the issue is not present in `v0.15.0`

**Comment by lukemaurer** [comment]
> Happy to provide the corresponding trace.

Yeah, if you want to attach the trace here, or email it to me at lmaurer@janestreet.com, I can have a look, thanks. Sadly the backtrace alone won't tell us much

**Comment by toots** [comment]
Ok, I emailed you the trace. Thanks!

**Comment by lukemaurer** [comment]
Got it. The trace loads on our current internal version (I do remember someone fixing a client-side crash recently), so this may be fixed in the janestreet-bleeding repo if you're willing to give that a go: https://github.com/janestreet/opam-repository

**Comment by toots** [comment]
Thanks. I didn't know about the repo! 

Unfortunately, it doesn't seem to compile in my `5.1.1` or `4.14.1` switches:

```
#=== ERROR while compiling js_of_ocaml_patches.v0.17~preview.129.17+77 ========#
# context     2.1.5 | macos/arm64 | ocaml-base-compiler.4.14.1 | git+https://github.com/janestreet/opam-repository.git
# path        ~/.opam/4.14.1/.opam-switch/build/js_of_ocaml_patches.v0.17~preview.129.17+77
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p js_of_ocaml_patches -j 11
# exit-code   1
# env-file    ~/.opam/log/js_of_ocaml_patches-2169-341334.env
# output-file ~/.opam/log/js_of_ocaml_patches-2169-341334.out
### output ###
# Error: The method ellipse has type
# [...]
#        but is expected to have type
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          Js_of_ocaml__.Js.number_t ->
#          bool Js_of_ocaml__.Js.t -> unit Js_of_ocaml__.Js.meth
#        Type unit Js_of_ocaml.Js.meth is not compatible with type
#          bool Js_of_ocaml__.Js.t -> unit Js_of_ocaml__.Js.meth


#=== ERROR while compiling base.v0.17~preview.129.17+77 =======================#
# context     2.1.5 | macos/arm64 | ocaml-base-compiler.4.14.1 | git+https://github.com/janestreet/opam-repository.git
# path        ~/.opam/4.14.1/.opam-switch/build/base.v0.17~preview.129.17+77
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p base -j 11
# exit-code   1
# env-file    ~/.opam/log/base-2169-b5b8b3.env
# output-file ~/.opam/log/base-2169-b5b8b3.out
### output ###
# [...]
# 37 |     with module Array := Shadow_stdlib.Array
# ...
# 40 |     with module Buffer := Shadow_stdlib.Buffer
# 41 |     with module Bytes := Shadow_stdlib.Bytes
# 42 |     with module Char := Shadow_stdlib.Char
# 43 |     with module Condition := Shadow_stdlib.Condition
# Error: Unbound module Shadow_stdlib.Condition
# (cd _build/default && /Users/toots/.opam/4.14.1/bin/ocamlopt.opt -w -40 -g -I src/.base.objs/byte -I src/.base.objs/native -I /Users/toots/.opam/4.14.1/lib/ocaml_intrinsics_kernel -I /Users/toots/.opam/4.14.1/lib/sexplib0 -I hash_types/src/.base_internalhash_types.objs/byte -I hash_types/src/.base_internalhash_types.objs/native -I shadow-stdlib/src/.shadow_stdlib.objs/byte -I shadow-stdlib/sr[...]
# File "src/array.ml", line 399, characters 28-75:
# 399 |       result := Bool.select ((keep_left_if [@inlined]) (compare x !result)) x !result
#                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Warning 55 [inlining-impossible]: Cannot inline: Unknown function
```

Any pointer as to how to compile it? A specific OCaml version perhaps?

**Comment by lukemaurer** [comment]
Hmm ... 4.14 should work, I would think. Actually, I'm told the fix for memtrace-viewer might not be out in bleeding after all quite yet. There should be a roll of bleeding later today, so worth trying again when that happens.

**Comment by dkalinichenko-js** [comment]
Hi! The `js_of_ocaml_patches` error looks like your version of `js_of_ocaml` does not match the one we test against (5.6.0). Could you provide yours so we can update the version bounds on the package? By the way, I pushed a fix for another issue with `memtrace_viewer` today, so you'll need to update the repo.

The bleeding edge version of Base requires 5.1, so it's expected that 4.14 fails.

**Comment by toots** [comment]
Ok I was able to install in a vanilla 5.1.1 switch! The latest `memtrace_viewer` is missing a dep on `codicons` though.

I am able to confirm that the issue is fixed in version `v0.17~preview.129.17+77`.

I was using version `5.6.0` of `js_of_ocaml`..

Thanks for the help!
